src/Form/inputTypeConfig.js

```diff
@@ -129,15 +130,11 @@ const inputTypeConfig = {
             const subtype = getInputSubtype(input)
             const variant = getInputVariant(input)
 
-            // Check first for password generation and the password.new scoring
-            if (device.settings.featureToggles.password_generation) {
-                if (subtype === 'password' && variant === 'new') {
-                    return canBeInteractedWith(input)
-                }
-            }
-
-            // if we are on a 'login' page, check if we have data to autofill the field
-            if (isLogin || isHybrid || variant === 'current') {
+            if (
+                (subtype === 'password' && variant === 'new') || // New passord field
+                (isLogin || isHybrid || variant === 'current') // Current password field
+            ) {
+                // Check feature flags and available input types
```

Comment:
I've moved the feature flag check to `canBeAutofilled` to keep concerns separate. This allowed me to simplify the checks here. I was almost able to consolidate all these functions to work the same, but then ran out of time.